frontend/src/app/main/ui/workspace/sidebar/assets.cljs

```diff
@@ -378,17 +378,21 @@
            (st/emit! (dw/unselect-all-assets))))
 
         on-component-click
+        (mf/use-fn
+         (mf/deps component selected-components)
+         (fn [event]
+           (dom/stop-propagation event)
+           (on-asset-click event (:id component) unselect-all)))
+
+        on-component-double-click
         (mf/use-fn
           (mf/deps component selected-components)
           (fn [event]
             (dom/stop-propagation event)
             (let [main-instance-id (:main-instance-id component)
                   main-instance-page (:main-instance-page component)]
-              (if (and main-instance-id main-instance-page)
-                (st/emit! (dw/go-to-main-instance main-instance-page main-instance-id
-                                                  #(on-asset-click event (:id component) unselect-all)))
-                ;; This may occur when :components-v2 is disabled
-                (on-asset-click event (:id component) unselect-all)))))
+              (when (and main-instance-id main-instance-page) ;; Only when :components-v2 is enabled
+                (st/emit! (dw/go-to-main-instance main-instance-page main-instance-id identity))))))
```

Comment:
Perhaps we should add another arity to this function without the parameter `on-page-selected`